dpctl-capi/source/dpctl_sycl_device_interface.cpp

```diff
@@ -612,11 +613,22 @@ DPCTLDevice_CreateSubDevicesByCounts(__dpctl_keep const DPCTLSyclDeviceRef DRef,
     vector_class<DPCTLSyclDeviceRef> *Devices = nullptr;
     std::vector<size_t> vcounts;
     vcounts.assign(counts, counts + ncounts);
-    auto D = unwrap(DRef);
-    if (D) {
+    size_t min_elem = *std::min_element(vcounts.begin(), vcounts.end());
+    if (DRef && min_elem) {
```

Comment:
Again some explanation to a user why they are getting back a nullptr will help.

Reply:
done